### PR TITLE
Using SSE4.2 intrinsics in basic_parser if available

### PR DESCRIPTION
**Pull request by octopus-prime**
issue #585

### CODE REVIEW

**Comment by octopus-prime** [comment]
Don't miss `-msse4.2` for gcc and clang. Otherwise the compiler will not generate SSE4.2 code ;-)

File: include/beast/http/detail/basic_parser.hpp [hunk 1]
```diff
@@ -215,7 +219,8 @@ class basic_parser_base
     {
         bool found = false;
 
-    #if ! BEAST_NO_INTRINSICS
+    #if (!BEAST_NO_INTRINSICS)
```
Comment (vinniefalco) [inline include/beast/http/detail/basic_parser.hpp]: Why the parenthesis?

File: include/beast/http/detail/basic_parser.hpp [hunk 2]
```diff
@@ -19,6 +19,10 @@
 #include <cstddef>
 #include <utility>
 
+#if (!BEAST_NO_INTRINSICS)
+#include <nmmintrin.h>
+#endif
```
Comment (vinniefalco) [inline include/beast/http/detail/basic_parser.hpp]: Is `<nmintrin.h>` available on msvc, gcc, clang?

**Comment by vinniefalco** [comment]
This is exciting!!

File: include/beast/http/detail/basic_parser.hpp (hunk 2 above)
Comment (octopus-prime) [inline include/beast/http/detail/basic_parser.hpp]: Yes, I think so.

File: include/beast/http/detail/basic_parser.hpp (hunk 1 above)
Comment (octopus-prime) [inline include/beast/http/detail/basic_parser.hpp]: Don't know. May be because of `if (...)` :D

**Comment by octopus-prime** [comment]
WTH! Builds failed because `fatal error: 'boost/predef/hardware/simd.h'` ?!???

**Comment by vinniefalco** [comment]
Welcome to the last year of my life.

Check the version of Boost used on travis:
https://github.com/vinniefalco/Beast/blob/develop/.travis.yml#L14

**Comment by vinniefalco** [comment]
I'm pulling the optimizations out of **v72** and putting them into **v73** so I can get the other branch merged

**Comment by vinniefalco** [comment]
>fatal error: 'boost/predef/hardware/simd.h' ?!???

Boost.Predef is really for end-users, not for writers of Boost libraries. Beast is supposed to use Boost.Config, in fact its mandatory as part of Boost.

**Comment by vinniefalco** [comment]
Hmm... GitHub closed the branch because the v72 branch was deleted...

**Pull request closed without merging**